### PR TITLE
Give Hubble access to StoreGetter and include ContainerStatuses field for Pod

### PR DESCRIPTION
**Pull request by michi-covalent**
this is a follow-up on #12114 with 3 commits:

1. Expose "pod" store in addition to "networkpolicy" and "namespace". 
2. Include containerStatuses field in PodStatus.
3. Include StoreGetter interface in CiliumDaemon.

### CODE REVIEW

**Comment by michi-covalent** [comment]
test-me-please

**Comment by michi-covalent** [comment]
test-me-please

**Comment by michi-covalent** [comment]
i'll trim a couple more fields that are not needed immediately.

**Comment by aanm** [comment]
> i'll trim a couple more fields that are not needed immediately.

I've marked the PR as draft while you are doing this.

**Comment by michi-covalent** [comment]
test-me-please

File: pkg/k8s/watchers/watcher.go [hunk 1]
```diff
@@ -786,6 +786,8 @@ func (k *K8sWatcher) GetStore(name string) cache.Store {
 		return k.networkpolicyStore
 	case "namespace":
 		return k.namespaceStore
+	case "pod":
+		return k.podStore
```
Comment (aanm) [inline pkg/k8s/watchers/watcher.go:795]:
```
	<-k.podStoreSet
	k.podStoreMU.RLock()
	defer k.podStoreMU.RUnlock()
        return k.podStore
```

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/watchers/watcher.go (hunk 1 above)
Comment (michi-covalent) [reply pkg/k8s/watchers/watcher.go:795]: @aanm i updated the PR. kind of curious why we need `k.podStoreMU`. looks like k.podStore gets set once during `podInit()` and never gets modified. isn't it sufficient to block on `k.podStoreSet`?

**Comment by michi-covalent** [comment]
test-me-please

**Comment by michi-covalent** [comment]
retest-net-next

File: pkg/k8s/watchers/watcher.go (hunk 1 above)
Comment (aanm) [reply pkg/k8s/watchers/watcher.go:795]: @michi-covalent with a kvstore setup + `enable-k8s-event-handover` the `podStore` variable can be set multiple times depending on the connectivity with the kvstore.

**Review by aanm — APPROVED** [review]

**Pull request merged**